### PR TITLE
UTILITIES: add utility that converts binary to MIPS assembly

### PR DESCRIPTION
**Pull request by DSsoto**

### CODE REVIEW

File: utilities/bin2ass/MIPSsim/MIPSsim.cpp [hunk 1]
```diff
@@ -0,0 +1,654 @@
...
+  }
+
+  bool operator()(string instruction, string instruction_type)
+  {
+    return func_map[instruction_type](instruction);
+  }
+
+  bool isCAT1(string inst)
+  {
+    return inst.substr(0, 3) == "001";
```
Comment (mattlangford) [inline utilities/bin2ass/MIPSsim/MIPSsim.cpp:67]: Are string comparisons more costly than converting strings to ints then comparing?

File: utilities/bin2ass/MIPSsim/MIPSsim.cpp [hunk 2]
```diff
@@ -0,0 +1,654 @@
...
+      if(isSW(inst))
+        return string{"SW"};
+      if(isLW(inst))
+        return string{"LW"};
+      if(isBREAK(inst))
+        return string{"BREAK"};
+    }
+    else if(isCAT2(inst))
+    {
+      if(isXOR(inst))
```
Comment (mattlangford) [inline utilities/bin2ass/MIPSsim/MIPSsim.cpp:271]: Wouldn't some kind of map be faster?

**Review by mattlangford — APPROVED** [review]
Nice work

File: utilities/bin2ass/MIPSsim/MIPSsim.cpp (hunk 1 above)
Comment (DSsoto) [reply utilities/bin2ass/MIPSsim/MIPSsim.cpp:67]: No, STL writers have implemented these kind of common comparisons with efficiency in mind, it's well optimized.

File: utilities/bin2ass/MIPSsim/MIPSsim.cpp (hunk 2 above)
Comment (DSsoto) [reply utilities/bin2ass/MIPSsim/MIPSsim.cpp:271]: I guess it could be.

**Pull request merged**